src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+
+import { parseBlastInput } from 'src/content/app/tools/blast/utils/blastInputParser';
+
+import BlastInputSequence from './BlastInputSequence';
+import PlusButton from 'src/shared/components/plus-button/PlusButton';
+
+import type { ParsedInputSequence } from 'src/content/app/tools/blast/types/parsedInputSequence';
+
+import styles from './BlastInputSequences.scss';
+
+const BlastInputSequences = () => {
+  const [sequences, setSequences] = useState<ParsedInputSequence[]>([]);
+  const [shouldAppendEmptyInput, setShouldAppendEmptyInput] = useState(false);
+
+  useEffect(() => {
+    if (!sequences.length) {
+      setShouldAppendEmptyInput(true);
+    }
+  });
+
+  const onSequenceAdded = (input: string, index: number | null) => {
+    const parsedSequences = parseBlastInput(input).filter((result) =>
+      Boolean(result.value)
+    ) as ParsedInputSequence[]; // <-- this would make a header without a sequence disappear; it is a good idea?
```

Comment:
I don't think it's a good idea. 

If I start by typing the header and then I try to copy and paste the sequence from somewhere else, I lose the header.

Reply:
Yes, I had similar apprehensions. I removed the filter. Probably should filter this out at submission step.

Reply:
✅